src/components/year_month.rs

```diff
@@ -2,14 +2,23 @@
 
 use std::str::FromStr;
 
+use tinystr::TinyAsciiStr;
+
 use crate::{
     components::calendar::Calendar,
     iso::{IsoDate, IsoDateSlots},
     options::ArithmeticOverflow,
+    utils::pad_iso_year,
     TemporalError, TemporalResult, TemporalUnwrap,
 };
 
-use super::calendar::GetTemporalCalendar;
+use super::{
+    calendar::{CalendarDateLike, GetTemporalCalendar},
+    Duration,
+};
+
+// Subset of `TemporalFields` representing just the  `YearMonthFields`
+pub struct YearMonthFields(pub i32, pub u8);
```

Comment:
Why do we need to expose the fields as public?
Also, I think this should be a proper struct with named fields for clarity